facade/baseFacade.go

```diff
@@ -263,31 +257,21 @@ func (epf *ElrondProxyFacade) SendUserFunds(receiver string, value *big.Int) err
 	return err
 }
 
-func (epf *ElrondProxyFacade) getNetworkConfig() (*networkConfig, error) {
-	netConfig, err := epf.nodeStatusProc.GetNetworkConfigMetrics()
+func (epf *ElrondProxyFacade) getNetworkConfig() (*data.NetworkConfig, error) {
+	genericResponse, err := epf.nodeStatusProc.GetNetworkConfigMetrics()
 	if err != nil {
 		return nil, err
 	}
 
-	netConf, ok := netConfig.Data.(map[string]interface{})["config"].(map[string]interface{})
-	if !ok {
-		return nil, errors.New("cannot get network config. something went wrong")
-	}
-
-	chainID, ok := netConf[core.MetricChainId].(string)
-	if !ok {
-		return nil, errors.New("cannot get chainID. something went wrong")
+	networkConfigBytes, err := json.Marshal(genericResponse.Data)
+	if err != nil {
+		return nil, err
 	}
 
-	version, ok := netConf[core.MetricMinTransactionVersion].(float64)
-	if !ok {
-		return nil, errors.New("cannot get version. something went wrong")
-	}
+	networkCfg := &data.NetworkConfig{}
+	err = json.Unmarshal(networkConfigBytes, networkCfg)
```

Comment:
Optional: see comment above. Returning a struct in `GetNetworkConfigMetrics` would be simpler.